zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -329,18 +331,72 @@ private static ContentDetailResolver getDefaultContentDetailResolver() {
         }
         if (StringUtils.isEmpty(session.getEmail())) {
-            throw new IllegalArgumentException("approval task unsuccesful: as session.email required but was null/empty");
+            throw new IllegalArgumentException(
+                    "approval task unsuccessful: as session.email required but was null/empty");
         }
-        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation sucessful");
+        info().data("collectionId", collection.getDescription().getId()).log("approval task: validation successful");
+    }
+
+    protected void uploadNewEndpoint(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        uploadWhitelistedFiles(collection, collectionReader);
+    }
+    
+    protected void uploadWhitelistedFiles(Collection collection, CollectionReader collectionReader) throws ZebedeeException, IOException {
+        for (String uri : collectionReader.getReviewed().listUris()) {
+            if (uri.endsWith(".csv") || uri.endsWith(".xlsx") || uri.endsWith(".xls") || uri.endsWith(".csdb")) {
+                String fileName = uri.substring(1);
+                Resource myFile = collectionReader.getResource(fileName);
+                if (DatasetWhitelistChecker.isWhitelisted(myFile.getName())) {
+                    info().log("File is whitelisted");
+                    uploadFile(myFile, fileName, collection.getDescription().getId());
+                } else {
+                    info().log("File is not whitelisted");
+                }
+            }
+        }
+    }
+    
+    protected void uploadFile(Resource myFile, String fileName, String collectionId) throws ZebedeeException, IOException {
+        File file = new File("afile");
+        try (FileOutputStream outputStream = new FileOutputStream(file)) {
+            IOUtils.copy(myFile.getData(), outputStream);
+        } catch (FileNotFoundException e) {
+            info().log("file not found");
+            throw e;
+        } catch (IOException e) {
+            info().log("input/output error");
+            throw e;
+        }
+        List<NameValuePair> params = createUploadParams(fileName, "path", collectionId);
+        uploadServiceSupplier.getService().uploadResumableFile(file, params);
+    }
+
+    protected static List<NameValuePair> createUploadParams(String resumableFilename, String path, String collectionId) {
+
+        // Get the following values from the config
+        String resumableType = Configuration.getResumableType();
+        String isPublishable = Configuration.getIsPublishable();
+        String licence = Configuration.getLicence();
+        String licenceURL= Configuration.getLicenceURL();
+
+        List<NameValuePair> params = new ArrayList<>();
+        params.add(new BasicNameValuePair("resumableFilename", resumableFilename));
+        params.add(new BasicNameValuePair("path", path));
```

Comment:
When I see the files in the bucket the key has the word "path" in the name (e.g , path/economy/grossdomesticproductgdp/datasets/mgdptest555/june23/mgdp.csdb) could the word path be removed please so the key would look like this economy/grossdomesticproductgdp/datasets/mgdptest555/june23/mgdp.csdb

Reply:
This is now addressed.

Reply:
No sorry, this is still in progress.

Reply:
This is now addressed.